backend/controller/scaling/k8sscaling/deployment_provisioner.go

```diff
@@ -141,6 +148,16 @@ func (r *DeploymentProvisioner) handleSchemaChange(ctx context.Context, msg *ftl
 				if err != nil {
 					logger.Errorf(err, "failed to delete deployment %s", msg.ModuleName)
 				}
+				// TODO: we only need to delete the services once this new ownership structure has been deployed to production
+				// Existing deployments don't have this though
+				logger.Infof("deleting service %s", msg.ModuleName)
+				err = r.Client.CoreV1().Secrets(r.Namespace).Delete(ctx, msg.DeploymentKey, v1.DeleteOptions{})
+				if err != nil {
+					if !errors.IsNotFound(err) {
+
+						logger.Errorf(err, "failed to delete service %s", msg.ModuleName)
```

Comment:
It's in a goroutine due to delayed deletion.

Reply:
In that case we should use an `errgroup.Group` or other mechanism to propagate the error, not drop it.

Reply:
There is a form of eventual consistency here, where 'orphaned' objects get cleaned up on the next schema change. If we used a wait group it would be the same as just adding a big sleep in the main thread.